Neos.Fusion/Classes/Core/ObjectTreeParser/ExceptionMessage/MessageLinePart.php

```diff
@@ -37,10 +37,10 @@ public function linePrint(int $offset = 0): string
 
     public function char(int $index = 0): string
     {
-        if ($index < 0) {
-            return mb_substr($this->linePart, $index, 1);
+        if (mb_strlen($this->linePart) < abs($index)) {
+            return '';
```

Comment:
I don't get this code here. From my POV it is a difference, if you use a `-4` or a `4` to a 4 char long string. So shouldn't be the handling also be different?
`'abcd'` => `-4` ==> `'a'`
`'abcd'` => `4` ==> `''`

Caused by the `0` index, which also counts for the positive indexes.

I assume your tests are running as `mb_substr` "fixes" this.

Reply:
yes this code was initially supposed to be an easy access on the string for building the exception message. But i messed it up a little :D Nothing api or anything further to worry about :D Thanks for looking into it though :D

Reply:
Ok, if the index < 0 exeeds than string start, it will always return the first char instead of empty string:
```
mb_substr(\"abcd\", -99, 1) == "a"
```

But index > 0 and index > lenght always returns ""
```
mb_substr(\"abcd\", 99, 1) == ""
```

That why an additional check for index < 0 is needed.